src/transport/SessionManager.h

```diff
@@ -566,6 +568,7 @@ class DLL_EXPORT SessionManager : public TransportMgrDelegate, public FabricTabl
     // Hold the TCPConnection callback context for the receiver application in the SessionManager.
     // On receipt of a connection from a peer, the SessionManager
     Transport::AppTCPConnectionCallbackCtxt * mServerTCPConnCbCtxt = nullptr;
+    Transport::PeerTCPParamsStorage mPeerTCPParamsStorage;
```

Comment:
This should probably not be hardcoding a specific storage strategy.  The way this sort of thing is normally done is:

1. Have a delegate API that is implementation-agnostic and exposes the high-level primitives that code actually depends on.
2. Have a default implementation of that API that uses a storage delegate as desired.

That allows SDK consumers to use a different implementation of the delegate API as needed.

In particular, clients must not require new "g/whatever" keys, in general, and this implementation here very much requires that.

Reply:
Ok. I wasn't fully aware of the dependence model for clients and thought this to be a core piece and part of the deeper transport layer and so directly implemented a concrete class. Going to separate into an Interface and Default Impl classes.

Reply:
Created an interface class and an implementation of it. The Server and Controller sides can be initialized with an implementation of that via the Init params.